app/models/grda_warehouse/tasks/sync_analysis_data_task.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# Sync data from the app db into the warehouse analysis data for export. It's necessary to pul
+# the data down and re-insert due to data boundary
+#
+# * This task can be removed if/when we unify the app and warehouse databases
+# * An alternative to this periodic sync would be a foreign data wrapper
+#
+module GrdaWarehouse::Tasks
+  class SyncAnalysisDataTask
+    def self.perform
+      new.perform
+    end
+
+    def perform
+      with_lock do
+        GrdaWarehouseBase.transaction do
+          sync_app_users
+          prune_removed_users
+        end
+      end
+    end
+
+    protected
+
+    def sync_app_users(batch_size: 500)
+      User.find_in_batches(batch_size: batch_size) do |batch|
```

Comment:
```suggestion
      User.with_deleted.find_in_batches(batch_size: batch_size) do |batch|
```
I _think_ we want to retain users that have been deleted, who may have data referencing them. That way for historical data its still possible to see the related staff user information.

I don't think users get deleted often (ever? it may not be possible in the UI?) but the model and schema allow it.